numalogic/models/autoencoder/base.py

```diff
@@ -29,6 +29,23 @@ def __init__(self, loss_fn: str = "huber", optim_algo: str = "adam", lr: float =
         self.optim_algo = optim_algo
         self.criterion = self.init_criterion(loss_fn)
 
+        self._total_train_loss = 0.0
+        self._total_val_loss = 0.0
+
+    @property
+    def total_train_loss(self):
+        return self._total_train_loss
+
+    @property
+    def total_val_loss(self):
+        return self._total_val_loss
+
+    def reset_train_loss(self):
+        self._total_train_loss = 0.0
+
+    def reset_val_loss(self):
```

Comment:
Any specific reason we are introducing these?

Reply:
Because before torch lightning 2.0 handled it automatically in the trainer. Now post 2.0, they are removing that, and giving more control to user so they have to define them. This was done to mostly speed up training and remove unnecessary overhead.